pytalibstream/atr_keeper.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+import pyslidingwindow
+from pytalibstream.types.hlc import HLC
+
+
+class AtrKeeper:
+    def __init__(self, max_len: Optional[int] = 10):
+        if max_len < 2:
+            raise Exception(
+                "AtrKeeper period must be >= 2")
+
+        self.high = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.low = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.close = pyslidingwindow.SlidingWindowFloat(max_len)
+        self._atr = 0
+        self.data_count = 0
+        self.max_len = max_len
+        self.prev_tr = []
+
+    def add(self, candle: HLC):
+        self.data_count += 1
+        self.high.push(candle.high)
+        self.low.push(candle.low)
+        self.close.push(candle.close)
+
+        if self.max_len + 1 > self.data_count > 1:
+            self.prev_tr.append(self.tr)
+        elif self.data_count == self.max_len+1:
+            self._atr = sum(self.prev_tr) / len(self.prev_tr)
+            self._atr = (self._atr * (self.max_len - 1) + self.tr) / self.max_len
+        else:
+            self._atr = (self._atr * (self.max_len - 1) + self.tr) / self.max_len
+
+    @staticmethod
+    def get_tr(high: float, low: float, close: float):
```

Comment:
the close is actually previous close. can we rename it to `prev_close`?

Reply:
see https://www.thebalance.com/how-average-true-range-atr-can-improve-trading-4154923#:~:text=Updated%20June%2009%2C%202020,of%20a%20stop%2Dloss%20order.